app/experimenter/experiments/api/v3/serializers.py

```diff
@@ -32,12 +32,17 @@ class Meta:
 
     def validate_name(self, name):
         slug = slugify(name)
-
-        if slug:
-            return name
-        raise serializers.ValidationError(
-            ["Name needs to contain alphanumeric characters"]
-        )
+        existing_slugs = Experiment.objects.values_list("slug", flat=True)
+
+        if not slug:
+            raise serializers.ValidationError(
+                ["Name needs to contain alphanumeric characters"]
+            )
+        if slug in existing_slugs:
```

Comment:
I think you can just `if Experiment.objects.filter(slug=slug).exists()` here instead of checking membership of the whole list?